### PR TITLE
add update setting to elasticsearch

### PR DESCRIPTION
**Pull request by jamesgorrie**
Seems to be fine. While ingesting it seemed fine as it updated in a very small amount of time, and given that we retry thrall's messages, this seemed full-proofish?

But! Pausing thrall might be the things we do prior, update the settings and then unpause.

Downtime? I suppose so.

### CODE REVIEW

**Comment by akash1810** [comment]
How would media-api react?

**Comment by jamesgorrie** [comment]
No read, no write.

**Comment by kenoir** [comment]
It's difficult to tell what the impact will be to the end user if this only takes a fraction of a second. My guess is that impact would be minimal. Given how onerous it is to perform a reindex (and the associated dangers of downtime) I'd say it's worth a try during a quiet period with comms sent out beforehand.

File: scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala [hunk 1]
```diff
@@ -143,6 +145,39 @@ object UpdateMapping extends EsScript {
...
+        indices.open(new OpenIndexRequest(imagesAlias))
+        client.close
+      }
+    }
+
+    EsClient.updateSettings
+  }
+
+  def usageError: Nothing = {
+    System.err.println("Usage: UpdateMapping <ES_HOST>")
```
Comment (theefer) [inline scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala]: Should it be `UpdateSettings` instead?

**Comment by theefer** [comment]
Are we confident this is going to be fast even on the PROD index that is an order of magnitude largest than TEST?

**Comment by theefer** [comment]
Could you describe what are we updating the settings to/for?

**Comment by jamesgorrie** [comment]
> Are we confident this is going to be fast even on the PROD index that is an order of magnitude largest than TEST?

I need to do some more discovery - as far as I can tell from the docs close, update, and open are just one off request. The concern I have is how long it would take that change to the setting to propagate to make sure we can open again.

File: scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala (hunk 1 above)
Comment (jamesgorrie) [inline scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala]: :eyes:

**Comment by jamesgorrie** [comment]
> Could you describe

This is a generic way to do it - do you mean in the PR?

**Comment by theefer** [comment]
I guess I am curious what are we going to use this for (that we didn't need until now)?

PR looks :+1: apart from being scary :-)

**Comment by jamesgorrie** [comment]
[We've added the hierarchy analyser](https://github.com/guardian/grid/blob/master/common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/Mappings.scala#L106-L111) to help search collections. We haven't been able to apply the mappings so far as we haven't updated the settings (including the analyser). So we need to update.

**Comment by theefer** [comment]
OK :+1:

**Comment by jamesgorrie** [comment]
Closing as the repercussions were to large in PROD.

**Pull request closed without merging**